### DISCUSSION TITLE
Issues with Coral running on i7-6700K with Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by yourpalfranc** [blakeblackshear/frigate#13916] [General]
I used the Getting Started guide and have my Frigate currently working successfully with 4 cameras (from a total of 12 that I have in my Blue Iris setup). I was unable to get the Coral drivers installed under Ubuntu 24.04 during the initial setup and several sources indicated that there are problems with that configuration. It was suggested to me in another post that I should work toward getting a stable Frigate setup with a few cameras and CPU, which I have done. I also tried using the Gasket Builder and the .deb file seemed to run fine, but the TPU remained unrecognized. I've spent hours googling for potential solutions with no success. So, where do I go from here? With only 4 cameras, my CPU is currently running at 70-ish% so adding any more cameras clearly is marginal. What's the issue with Ubuntu 24.04? Is it a problem with a hope of being resolved, or is there possibly a different distro that would better support the Frigate/Coral configuration. Thanks in advance!

**Comment by NickM-27** [comment]
if you have an i7-6700K then you can just use OpenVINO https://docs.frigate.video/configuration/object_detectors#openvino-detector

**Reply by yourpalfranc** [reply]
I will definitely look into OpenVINO to improve things, but what's up with Coral?? I mean, it's HIGHLY recommended hardware in the Frigate docs??? Thanks for the suggestion.

**Reply by NickM-27** [reply]
I don't have much experience with Ubuntu so I can't really help there, I've never had a problem with the coral using Debian as the host os (which is what the frigate docs recommend)

**Reply by yourpalfranc** [reply]
Ubuntu is a Debian derivative, but it's not pure Debian, I guess...and it was recommended in the Getting Started pdf file that I used...yes, I know that's not official Frigate documentation. But, I'll look into OpenVINO if another solution doesn't work to install the Coral. Thanks!!

**Reply by yourpalfranc** [reply]
I'm beginning to wonder if it might be the best move for me to start from scratch with a (straight) Debian install. Are you using the Debian 12 download from debian.org? I don't have any experience with Debian, but since Ubuntu is a derivative, I should know it at a fundamental level, and since it appears to be the development platform for Frigate, it probably makes more sense to run it that way.

**Reply by NickM-27** [reply]
the docs walk you through step by step https://docs.frigate.video/guides/getting_started

**Reply by yourpalfranc** [reply]
I think I'll just do that. It will probably save me a bunch of time and effort in the long run, plus I'm assuming that since I have a working config all the effort to get my initial camera setup going will be avoided by saving and restoring the config.yml file to the new Debian 12 setup. Thanks!

**Reply by yourpalfranc** [reply]
Is the i386 package the correct one for i7-6700K?

**Reply by NickM-27** [reply]
No, amd64

**Reply by yourpalfranc** [reply]
Thanks.

**Reply by yourpalfranc** [reply]
I've run into my first Debian roadblock. I've reached step 5 in the Getting Started - Install and setup Debian for remote access. Next step (i) install sudo ran fine, but step (ii) returns "bash: usermod: command not found" and yes, I did replace my username for blake. I'm really unfamiliar with running at root level, so I have some learning to do there.

**Reply by NickM-27** [reply]
usermod is most likely installed, according to https://www.reddit.com/r/linuxquestions/comments/pcfjo6/comment/haijm8f/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button you should be able to run `su -` and then run the usermod command

**Reply by yourpalfranc** [reply]
Yes, I was just about to tell you that I'd neglected to add the hyphen to su. Seems to be fine now. Sorry for the inconvenience.

**Reply by yourpalfranc** [reply]
Looking good so far. I have Frigate up and running with one camera and I've reached a point where I need to stop on this project for the day. I like it better already...running remotely and all the PC resources are devoted to Frigate. I'll work on completing the Getting Started steps and (hopefully) adding the Coral tomorrow. I sure appreciate your help!!

**Reply by yourpalfranc** [reply]
Me again...I'm hitting a wall with the Coral installation. When running the second step in the Coral docs, I get the following return...

fcromer@FrigateNVR:\~$ sudo apt-get install gasket-dkms libedgetpu1-std
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package gasket-dkms
E: Unable to locate package libedgetpu1-std
fcromer@FrigateNVR:\~$

**Reply by NickM-27** [reply]
you probably need to run `sudo apt-get update` first

**Reply by yourpalfranc** [reply]
Same result.

**Reply by NickM-27** [reply]
so to be clear you ran

```
echo "deb https://packages.cloud.google.com/apt coral-edgetpu-stable main" | sudo tee /etc/apt/sources.list.d/coral-edgetpu.list

curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -

sudo apt-get update
```

**Reply by yourpalfranc** [reply]
Yes, I copy/pasted the command directly from the Coral docs, but it didn't work. I've been getting some assistance from Coral support and apparently it may have been an issue with a missing public key. After correcting that, it appeared that the driver may have downloaded and installed, but a hardware issue reared its ugly head...it appears that the PCIe adapter that I used may somehow be incompatible with the Dell Precision Tower PC I'm working with even though I'm using the exact same adapter in another Dell PC that's running my Blue Iris setup perfectly. The Dell Precision Tower just won't boot with the PCIe adapter/Coral TPU installed. I've ordered a different adapter to see if it rectifies the issue, so I'm still I'm still in limbo until it arrives. Thanks for your reply.

**Comment by moandcompany** [comment]
There's a few ways to address the pycoral  library installation issues you're encountering on Ubuntu 22.04+ or later.

1. https://igor.technology/installing-coral-usb-accelearator-python-3-10-ubuntu-22/
2. https://github.com/google-coral/pycoral/issues/85#issuecomment-2282225345

Recommend trying to use the prebuilt wheel (1) before compiling your own (2).

Also, a simpler approach to running Frigate would be via the prebuilt Docker images. Pass through the USB or PCIe Coral Edge TPU to your container:
https://docs.frigate.video/frigate/installation/#docker

**Reply by yourpalfranc** [reply]
I don't have experience with wheel files, but these instructions look fairly straightforward. I'll give it a go tomorrow.

I did actually use these Docker installation instructions...same docker-compose.yaml script.

Thanks for the help. I'll let you know how I faired with the wheel files.

**Reply by yourpalfranc** [reply]
The process you've suggested as your first solution identifies specifically with the Coral USB. Is it equally applicable to the Coral PCIe? Just going by the Coral docs there seems to be a difference.

**Reply by NickM-27** [reply]
no, it is different

**Comment by moandcompany** [comment]
When using some of the older (e.g. Skylake - 6xxx) Intel integrated GPUs for hardware accelerated video decoding as well as model inference  (i.e. detection) via OpenVINO, you may find some system instability issues.

- https://github.com/blakeblackshear/frigate/issues/8338

**Reply by yourpalfranc** [reply]
Thanks. I'm really more hopeful of getting the Coral installed. I'm currently using one in the Blue Iris system that I'm working to replace with Frigate, and it performs great. I'll be working on the solution you  provided later this afternoon. Thanks again.

**Comment by FabianGabor** [comment]
I used the [official install guide](https://coral.ai/docs/accelerator/get-started/#1-install-the-edge-tpu-runtime) for Ubuntu 24 and worked flawlessly.

**Reply by yourpalfranc** [reply]
I have the Coral PCIe so I used the official Coral install guide for that hardware, and unfortunately, I didn't have the same successful result you had.

**Comment by djieno** [comment]
I couldn't get the apt version to work using the offical install guide but building it [did work](https://github.com/google-coral/edgetpu/issues/866#issuecomment-2417201658).

**Reply by yourpalfranc** [reply]
Thanks for the reply. I finally got it working with some guidance from Coral support. Seems I was missing some keys.

**Reply by moogoos** [reply]
> Thanks for the reply. I finally got it working with some guidance from Coral support. Seems I was missing some keys.

Thanks for sharing exactly what..
